submissions/EmmanuelIdiaghe/slim-php/publishComp/app/controllers/ArticleController.php

```diff
@@ -0,0 +1,111 @@
+                'created_at' => date('Y-m-d H:m:s')
+            ]);
+            $article->save();
+            $response->getBody()->write($article->toJson());
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(200);
+        }
+        else {
+            $response->getBody()->write(json_encode(['status' => 'failure', 'message' => 'Empty input fields']));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(400);
+        }
+    }
+
+    //2. Update existing article
+    public function updateArticle(Request $request, Response $response, $args) {
+        $id = $args['id'];
+        $data = $request->getParsedBody();
+
+        $article = Article::find($id);
+        if ($article) {
+            $article->title = $data['title'];
+            $article->save();
+            $response->getBody()->write($article->toJson());
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(200);
+        }
+        else {
+            $response->getBody()->write(json_encode(['status' => 'failure', 'message' => 'This article does not exist!']));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(404);
+        }
+    }
+
+    //3. Delete existing article
+    public function deleteArticle(Request $request, Response $response, $args) {
+        $id = $args['id'];
+
+        $article = Article::find($id);
+        if ($article) {
+            $article->delete();
+            $response->getBody()->write(json_encode(['status' => 'success', 'message' => 'This article was successfully deleted']));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(200);
+        }
+        else {
+            $response->getBody()->write(json_encode(['status' => 'failure', 'message' => 'This article does not exist!']));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(404);
+        }
+    }
+
+    //4. Update existing article
+    public function publishArticle(Request $request, Response $response, $args) {
+        $id = $args['id'];
+        $data = $request->getParsedBody();
+
+        $article = Article::find($id);
+        if ($article) {
+            $article->status_ = 'published';
+            $article->save();
+            $response->getBody()->write(json_encode(['status' => 'success', 'data' => $article]));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(200);
+        }
+        else {
+            $response->getBody()->write(json_encode(['status' => 'failure', 'message' => 'This article does not exist!']));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(404);
+        }
+    }
+
+    //5. Show all published articles
+    public function allArticles(Request $request, Response $response) {
+        $articles = Article::all();
```

Comment:
Well done. But, this will return all articles on the table, whether the status is published or not. You are expected to return all `published` articles. Meaning you have to use a `WHERE` clause in your query